### DISCUSSION TITLE
Rename tabs

### DISCUSSION THREAD

**Discussion by Tobbe** [wezterm/wezterm#1048] [Q&A] [answered]
Hi

Is it possible to create a key binding to something that'd allow me to rename the current tab?
Like in another terminal I'm using I have LCtrl+R pop up a dialog where I can enter a new name for the active tab.

I work on a few different services/libraries so what I like to do when I'm going to work on one of them is to open two or three new tabs, one for regular cli work, like file management, git, grep, etc. And another for a long-running process like the dev server, and sometimes a third for automatic unit tests etc. And then I like to name those tabs "libA", "libA srv" and "libA test". It's not uncommon for me to have 8+ tabs open for 4 or 5 different services or libs. And all of them would be in /C/Users/tobbe/dev/company/libraryXYZ where only the last part differs, and where the path is always exactly the same for two or three tabs.

(I don't always need to have all servers etc running, so it'd also be useful to be able to indicate what tabs have a process running and/or have new output. Another terminal I use do this by appending an asterisk (*) to the end of the tab title)

Here's what my tab bar currently looks like, with only 7 tabs

![image](https://user-images.githubusercontent.com/30793/130054867-89eb0a2c-26c0-49d4-87b7-98230508ce6b.png)

**Comment by wez** [comment, marked as answer]
Thanks for the suggestion (and the suggested key binding).  This feature request is already being tracked in https://github.com/wez/wezterm/issues/522

**Reply by Tobbe** [reply]
Thanks for the link :)  Found a couple of other issues tagged as "Enhancement" that I'm looking forward to as well :)

**Reply by kar1** [reply]
fwiw check my answer here https://github.com/wez/wezterm/issues/522#issuecomment-902203635
